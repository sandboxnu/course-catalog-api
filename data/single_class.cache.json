{
    "neu": {
        "classes": [
            {
                "host": "neu.edu",
                "termId": "202240",
                "subject": "CS",
                "classId": "2510",
                "classAttributes": [
                    "NUpath Analyzing/Using Data  NCAD",
                    "NUpath Natural/Designed World  NCND",
                    "Computer&Info Sci  UBCS"
                ],
                "nupath": [
                    "Analyzing/Using Data",
                    "Natural/Designed World"
                ],
                "desc": "Continues CS 2500. Examines object-oriented programming and associated algorithms using more complex data structures as the focus. Discusses nested structures and nonlinear structures including hash tables, trees, and graphs. Emphasizes abstraction, encapsulation, inheritance, polymorphism, recursion, and object-oriented design patterns. Applies these ideas to sample applications that illustrate the breadth of computer science.",
                "prettyUrl": "https://wl11gp.neu.edu/udcprod8/bwckctlg.p_disp_course_detail?cat_term_in=202240&subj_code_in=CS&crse_numb_in=2510",
                "name": "Fundamentals of Computer Science 2",
                "url": "https://wl11gp.neu.edu/udcprod8/bwckctlg.p_disp_listcrse?term_in=202240&subj_in=CS&crse_in=2510&schd_in=%",
                "lastUpdateTime": 1644620452425,
                "maxCredits": 4,
                "minCredits": 4,
                "college": "NEU",
                "feeAmount": null,
                "feeDescription": "",
                "prereqs": {
                    "type": "and",
                    "values": [
                        {
                            "classId": "2500",
                            "subject": "CS"
                        }
                    ]
                },
                "coreqs": {
                    "type": "and",
                    "values": [
                        {
                            "classId": "2511",
                            "subject": "CS"
                        }
                    ]
                },
                "optPrereqsFor": {
                    "values": [
                        {
                            "subject": "CS",
                            "classId": "3000"
                        },
                        {
                            "subject": "CS",
                            "classId": "3500"
                        },
                        {
                            "subject": "CS",
                            "classId": "3520"
                        },
                        {
                            "subject": "CS",
                            "classId": "3650"
                        },
                        {
                            "subject": "CS",
                            "classId": "3800"
                        },
                        {
                            "subject": "CS",
                            "classId": "4300"
                        },
                        {
                            "subject": "DS",
                            "classId": "3000"
                        },
                        {
                            "subject": "EECE",
                            "classId": "2322"
                        }
                    ]
                },
                "prereqsFor": {
                    "values": []
                }
            }
        ],
        "subjects": {
            "ACCT": "Accounting",
            "ACC": "Accounting - CPS",
            "AVM": "Adv Manufacturing System - CPS",
            "AFAM": "African American Studies",
            "AFRS": "African Studies",
            "ASL": "American Sign Lang - CPS",
            "AMSL": "American Sign Language",
            "ALY": "Analytics - CPS",
            "ANTH": "Anthropology",
            "ANT": "Anthropology - CPS",
            "ARAB": "Arabic",
            "ARCH": "Architecture",
            "ARMY": "Army ROTC",
            "ART": "Art - CPS",
            "ARTG": "Art - Design",
            "ARTF": "Art - Fundamentals",
            "ARTE": "Art - General",
            "ARTH": "Art - History",
            "ARTD": "Art - Media Arts",
            "ARTS": "Art - Studio",
            "AACE": "Arts Admin & Cultural Entrep",
            "ASNS": "Asian Studies",
            "BNSC": "Behavioral Neuroscience",
            "BIOC": "Biochemistry",
            "BIOE": "Bioengineering",
            "BINF": "Bioinformatics",
            "BIOL": "Biology",
            "BIO": "Biology - CPS",
            "BIOT": "Biotechnology",
            "BTC": "Biotechnology - CPS",
            "BUSN": "Business Administration",
            "EXSC": "Cardiopulmonary & Exercise Sci",
            "CDV": "Career Development - CPS",
            "CHME": "Chemical Engineering",
            "CHEM": "Chemistry & Chemical Biology",
            "CHM": "Chemistry - CPS",
            "CHNS": "Chinese",
            "CIVE": "Civil & Environmental Engineer",
            "CRS": "Cnsl Psy/Rehab/Sp Ed - CPS",
            "CED": "Commerce & Economic Dev - CPS",
            "COMM": "Communication Studies",
            "CMN": "Communicatn Studies - CPS",
            "CMMN": "Communicatn Studies - CPS Spec",
            "ETC": "Comp Engneering Tech - CPS",
            "CET": "Computer Engineerng Tech - CPS",
            "CS": "Computer Science",
            "CSYE": "Computer Systems Engineering",
            "CMG": "Construction Mangmnt - CPS",
            "EEAM": "Coop/Exper Ed - Arts/Med/Dsgn",
            "EESC": "Coop/Exper Ed - Science",
            "EESH": "Coop/Exper Ed - Soc Sci/Hum",
            "COOP": "Cooperative Education",
            "COP": "Cooperative Educatn - CPS",
            "CAEP": "Counseling and Applied Ed Psyc",
            "CRIM": "Criminal Justice",
            "CJS": "Criminal Justice - CPS",
            "CLTR": "Culture",
            "LITR": "Culture - Literature",
            "CY": "Cybersecurity",
            "DA": "Data Analytics",
            "DAMG": "Data Architecture Management",
            "DS": "Data Science",
            "DEAF": "Deaf Studies",
            "DGM": "Digital Media - CPS",
            "GEO": "Earth & Env Sci - CPS",
            "ENVR": "Earth & Environmental Sciences",
            "EEMB": "Ecology, Evolutn & Marine Biol",
            "ECON": "Economics",
            "ECN": "Economics - CPS",
            "ECNM": "Economics - CPS Spec",
            "EDUC": "Education",
            "EDU": "Education - CPS",
            "EECE": "Electrical and Comp Engineerng",
            "EET": "Electrical Engineer Tech - CPS",
            "ENSY": "Energy Systems",
            "ENCP": "Engineering Cooperative Ed",
            "ENGR": "Engineering Interdisciplinary",
            "ENLR": "Engineering Leadership",
            "EMGT": "Engineering Management",
            "ENGL": "English",
            "ENG": "English - CPS",
            "ESL": "English as Scnd Lang - CPS",
            "ESLG": "English as Scnd Lng - CPS Spec",
            "ENGW": "English Writing",
            "EAI": "Enterprise Artfcal Intellgnce",
            "ENTR": "Entrepreneurship & Innovation",
            "TECE": "Entrepreneurship Technological",
            "ESC": "Environmental Science - CPS",
            "ENVS": "Environmental Studies",
            "FINA": "Finance & Insurance",
            "FIN": "Finance - CPS",
            "FSEM": "First-Year Seminar",
            "FLNG": "Foreign Language",
            "FRNH": "French",
            "FRN": "French - CPS",
            "GAME": "Game Design",
            "GSND": "Game Science and Design",
            "GEN": "General - CPS",
            "GE": "General Engineering",
            "GET": "General Engineering Tech - CPS",
            "GENS": "General Studies",
            "GIS": "Geographic Info Sys - CPS",
            "GRMN": "German",
            "GST": "Global Studies - CPS",
            "GBST": "Global Studies - CPS Spec",
            "GSE": "Graduate Engineering - CPS",
            "GREK": "Greek",
            "HIA": "Health Info Admin - CPS",
            "HINF": "Health Informatics",
            "HMG": "Health Management - CPS",
            "HLTH": "Health Sci - Interdisciplinary",
            "HSCI": "Health Science",
            "HSC": "Health Science - CPS",
            "HBRW": "Hebrew",
            "HIST": "History",
            "HST": "History - CPS",
            "HSTY": "History - CPS Spec",
            "HLS": "Homeland Security - CPS",
            "HONR": "Honors Program",
            "HRMG": "Human Resources Management",
            "HRM": "Human Resources Mgmt - CPS",
            "HUSV": "Human Services",
            "HSV": "Human Services - CPS",
            "HUM": "Humanities - CPS",
            "IE": "Industrial Engineering",
            "IRM": "Info Resources Mgmt - CPS",
            "IS": "Information Science",
            "INFO": "Information Systems Program",
            "ITC": "Information Tech - CPS",
            "ITP": "Intellectual Proprty - CPS",
            "INAM": "Interdisc Studies - Arts/Media",
            "INPR": "Interdisc Studies - Provost",
            "INSC": "Interdisc Studies - Science",
            "INSH": "Interdisc Studies - Soc Sc/Hum",
            "INT": "Interdiscpln Studies - CPS",
            "INTL": "International Affairs",
            "INTB": "International Business",
            "IEAR": "Internatl Ed - Arts",
            "IEBN": "Internatl Ed - Business",
            "IECS": "Internatl Ed - Computer Sci",
            "IEHS": "Internatl Ed - Health Sciences",
            "IEHM": "Internatl Ed - Humanities",
            "IELG": "Internatl Ed - Language",
            "IESC": "Internatl Ed - Science",
            "IESO": "Internatl Ed - Social Sci",
            "INTP": "Interpreting",
            "ITLN": "Italian",
            "JPNS": "Japanese",
            "JWSS": "Jewish Studies",
            "JLS": "Jewish Studies - CPS",
            "JRNL": "Journalism",
            "KMG": "Knowledge Mgmt - CPS",
            "LARC": "Landscape Architecture",
            "LANG": "Language - General",
            "LACS": "Latin Am & Carib Studies",
            "LAW": "Law",
            "LWP": "Law & Policy - CPS",
            "LW": "Law (for Non-Law School Stu)",
            "LPSC": "Law and Public Policy",
            "LDR": "Leadership Studies - CPS",
            "LS": "Legal Studies",
            "XLVL": "Level Bootcamp",
            "LST": "Liberal Studies - CPS",
            "LING": "Linguistics",
            "MGMT": "Management",
            "MGT": "Management - CPS",
            "MIS": "Management Info Sys - CPS",
            "MISM": "Management Information Systems",
            "MGSC": "Management Science",
            "MECN": "Managerial Economics",
            "MKTG": "Marketing",
            "MKT": "Marketing - CPS",
            "MATL": "Materials Engineering",
            "MATH": "Mathematics",
            "MTH": "Mathematics - CPS",
            "MATM": "Mathematics - CPS Spec",
            "MEIE": "Mech & Industrial Engineering",
            "MET": "Mechanical Engineer Tech - CPS",
            "ME": "Mechanical Engineering",
            "CINE": "Media - Cinema Studies",
            "MSCR": "Media and Screen Studies",
            "LNG": "Modern Languages - CPS",
            "MUSC": "Music",
            "MUS": "Music - CPS",
            "MSIC": "Music - CPS Spec",
            "MUSI": "Music Industry",
            "MPNC": "Music Performance - NEC",
            "MUST": "Music Technology",
            "NNMD": "Nanomedicine",
            "NETS": "Network Science",
            "NPM": "Nonprofit Management - CPS",
            "NRSG": "Nursing",
            "NTR": "Nutrition - CPS",
            "OR": "Operations Research",
            "ORGB": "Organizational Behavior",
            "OCM": "Organizational Change - CPS",
            "PHSC": "Pharmaceutical Science",
            "PMST": "Pharmaceutics",
            "PMCL": "Pharmacology",
            "PHMD": "Pharmacy Practice",
            "PMC": "Pharmacy-Med Chem - CPS",
            "PHDL": "PhD Experiential Leadership",
            "PHIL": "Philosophy",
            "PHL": "Philosophy - CPS",
            "PHLS": "Philosophy - CPS Spec",
            "RELS": "Philosophy - Religious Studies",
            "PHE": "Physical Education - CPS",
            "PT": "Physical Therapy",
            "PTH": "Physical Therapy - CPS",
            "PA": "Physician Assistant",
            "PHYS": "Physics",
            "PHY": "Physics - CPS",
            "POLS": "Political Science",
            "POL": "Political Science - CPS",
            "PLSC": "Political Science - CPS Spec",
            "PORT": "Portuguese",
            "PDP": "Prof Devlpmnt Progs - CPS",
            "PJM": "Project Management - CPS",
            "PSYC": "Psychology",
            "PSY": "Psychology - CPS",
            "PPUA": "Pub Policy and Urban Affairs",
            "PHTH": "Public Health",
            "PBR": "Public Relations - CPS",
            "RGA": "Regulatory Affairs - CPS",
            "RFA": "Regulatory Affairs Food - CPS",
            "RMS": "Remote Sensing - CPS",
            "RPT": "Respiratory Therapy - CPS",
            "RSSN": "Russian",
            "SMFA": "School of Museum of Fine Arts",
            "SCI": "Science - CPS",
            "SOCL": "Sociology",
            "SOC": "Sociology - CPS",
            "SCLY": "Sociology - CPS Spec",
            "SPNS": "Spanish",
            "SPN": "Spanish - CPS",
            "SLPA": "Speech-Lang Path & Audiology",
            "SIA": "Strategic Intel/Analysis - CPS",
            "STRT": "Strategy",
            "ABRD": "Study Abroad",
            "ABRB": "Study Abroad - Business",
            "ABRC": "Study Abroad - CPS Spec",
            "ABRL": "Study Abroad - Law",
            "ABRS": "Study Abroad - Science",
            "ABRH": "Study Abroad - Soc Sci & Hum",
            "ABRV": "Study Abroad - Virtual",
            "ABRU": "Study USA",
            "SCHM": "Supply Chain Management",
            "SBSY": "Sustainable Building Systems",
            "SUEN": "Sustainable Urban Environments",
            "SWHL": "Swahili",
            "TCM": "Tech Commercialization - CPS",
            "TCC": "Technical Communic - CPS",
            "TELR": "Technology Leadership",
            "TELE": "Telecommunication Systems",
            "THTR": "Theatre",
            "TOXC": "Toxicology",
            "URBS": "Urban Studies",
            "WMNS": "Women's/Gender/Sexualty Stdies"
        }
    }
}