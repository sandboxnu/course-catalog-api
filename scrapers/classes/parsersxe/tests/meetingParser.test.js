import MeetingParser, { forTesting } from "../meetingParser";
import data from "./data/meetingParser.data";

const { hhmmToSeconds, mmddyyyyToDaysSinceEpoch } = forTesting;

it("meetingParser.ts", () => {
  expect(MeetingParser.parseMeetings(data.htlh2200)).toMatchSnapshot();
});

it("profname", () => {
  expect(MeetingParser.profName({ displayName: "Chu, Daj" })).toEqual(
    "Daj Chu",
  );
  expect(MeetingParser.profName("Lastname, Firstname")).toEqual(
    "Firstname Lastname",
  );
});

it("hhmmToSeconds", () => {
  expect(hhmmToSeconds("0000")).toEqual(0);
  expect(hhmmToSeconds("0010")).toEqual(600);
  expect(hhmmToSeconds("0100")).toEqual(3600);
  expect(hhmmToSeconds("0101")).toEqual(3660);
});
