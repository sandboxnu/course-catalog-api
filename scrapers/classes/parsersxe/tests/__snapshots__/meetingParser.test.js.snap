// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`meetingParser.ts 1`] = `
[
  {
    "endDate": 18067,
    "startDate": 18022,
    "times": {
      "2": [
        {
          "end": 48000,
          "start": 35400,
        },
      ],
      "4": [
        {
          "end": 48000,
          "start": 35400,
        },
      ],
      "5": [
        {
          "end": 48000,
          "start": 35400,
        },
      ],
    },
    "type": "Class",
    "where": "Forsyth Building 242",
  },
  {
    "endDate": 18072,
    "startDate": 18022,
    "times": {
      "1": [
        {
          "end": 52200,
          "start": 36000,
        },
      ],
      "3": [
        {
          "end": 52200,
          "start": 36000,
        },
      ],
    },
    "type": "Lab",
    "where": "Behrakis Health Sciences Cntr 420",
  },
  {
    "endDate": 18072,
    "startDate": 18072,
    "times": {
      "2": [
        {
          "end": 63000,
          "start": 55800,
        },
      ],
    },
    "type": "Final Exam",
    "where": "Cargill Hall 094",
  },
]
`;
